src/imap/mod.rs

```diff
@@ -295,18 +298,40 @@ impl Imap {
                 // needs to be set here to ensure it is set on reconnects.
                 self.connected = true;
                 self.session = Some(session);
+                self.login_failed_once = false;
                 Ok(())
             }
+
             Err((err, _)) => {
                 let imap_user = self.config.imap_user.to_owned();
                 let message = context
                     .stock_string_repl_str(StockMessage::CannotLogin, &imap_user)
                     .await;
 
-                emit_event!(
-                    context,
-                    Event::ErrorNetwork(format!("{} ({})", message, err))
-                );
+                warn!(context, "{} ({})", message, err);
+                emit_event!(context, Event::ErrorNetwork(message.clone()));
+
+                let lock = context.wrong_pw_warning_mutex.lock().await;
+                if self.login_failed_once
+                    && context.get_config_bool(Config::NotifyAboutWrongPw).await
+                {
+                    if let Err(e) = context.set_config(Config::NotifyAboutWrongPw, None).await {
+                        warn!(context, "{}", e);
+                    }
+                    drop(lock);
+
+                    let mut msg = Message::new(Viewtype::Text);
+                    msg.text = Some(message);
+                    if let Err(e) =
+                        chat::add_device_msg_with_importance(context, None, Some(&mut msg), true)
+                            .await
+                    {
+                        warn!(context, "{}", e);
+                    }
+                } else {
+                    self.login_failed_once = true;
```

Comment:
I don't see why `login_failed_once` is needed. Why not notify when it fails for the first time?

Reply:
My thought on this was that there might be other reasons for the login failing, like the internet being gone. But if the connection failed twice in a row when trying to log in then it is likely that it is because the password is wrong.

Reply:
wondering, if there is no error-code or so that we can check here. if this is not possible or reliable, this "trick" is not that bad: if internet is gone, next attempt would probably also fail some lines before.

Reply:
I think that there is a message like "Authentication failed" sent but I feared that some server would change the wording slightly or something.